src/lib/THERMAL/thermal.cpp

```diff
@@ -17,18 +20,32 @@ uint8_t thermal_threshold_data[] = {
 };
 
 int thermal_status = THERMAL_STATUS_FAIL;
+static bool use_internal_sensor = false;
 
 void Thermal::init()
 {
     int status = -1;
-#ifdef HAS_THERMAL_LM75A
-    status = lm75a.init();
-#endif
+    if (OPT_HAS_THERMAL_LM75A)
+    {
+        status = lm75a.init();
+    }
+#if defined(PLATFORM_ESP32_S3)
+    if (status == -1)
+    {
+        temp_sensor_config_t temp_sensor = TSENS_CONFIG_DEFAULT();
+        temp_sensor.dac_offset = TSENS_DAC_L2;  //TSENS_DAC_L2 is default   L4(-40℃ ~ 20℃), L2(-10℃ ~ 80℃) L1(20℃ ~ 100℃) L0(50℃ ~ 125℃)
+        temp_sensor_set_config(temp_sensor);
+        temp_sensor_start();
+        use_internal_sensor = true;
+    }
+#else
     if(status == -1)
     {
         ERRLN("Thermal failed!");
+        return;
```

Comment:
I like the return, can you remove the else and de-indent the Thermal OK code too so it is clear it always is supposed to run?

Reply:
Yes, that makes complete sense! What was I thinking...